### ISSUE TITLE
[bug]: accept-encoding gzip does not decode

### ISSUE THREAD

**Issue by Marcuse7** [hoppscotch/hoppscotch#4387]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When I send a GET request with header `accept-encoding: gzip, deflate` I expect the response to be decoded. What I get is undecoded (gzipped) text.

### Steps to reproduce

1. Open URL https://hoppscotch.io/
2. Use default `GET` query with https://echo.hoppscotch.io 
3. Add header `Accept-encoding` value `gzip` or `gzip, deflate`
4. Execute query

### Environment

Production

### Version

Cloud

**Comment by ipaulaandreea** [comment]
I would like to work on this

**Comment by Marcuse7** [comment]
Hello Vansh, thank you for reply!
I use Hoppscotch Version 24.9.1-0 (20241006.121003) on macOS 14.7.
This is the request and its reply:
![Screenshot 2024-10-13 at 14 00 55](https://github.com/user-attachments/assets/fdfd3471-9818-4742-8b8d-36d0127f767c)
These are my settings (the error persists w/ interceptor set to Native):
![Screenshot 2024-10-13 at 14 02 59](https://github.com/user-attachments/assets/8e509dcc-21ba-42e5-9ce5-3dfd94eece47)

**Comment by unacro** [comment]
Same situation.

While response body is zipped, show RAW content as garble
(header `content-encoding` marked as `gzip`)

**Comment by AugleBoBaugles** [comment]
I'm working on this

**Comment by SanskritiHarmukh** [comment]
Hey @AugleBoBaugles , just wanted to check if you're participating in the CodeDay micro-internship program, as issues labeled with `CodeDay` are reserved for that partnership. Let me know!

**Comment by AugleBoBaugles** [comment]
Yes, this is the issue assigned to my group for the CodeDay micro-internship.

**Comment by AugleBoBaugles** [comment]
@SanskritiHarmukh My team has hit a wall with this issue. We’ve successfully set up our build environments, however we haven’t been able to recreate the issue locally. We have been able to recreate it using the current version of Hoppscotch at https://hoppscotch.io/ though. 

We’ve seen that there was an update to the Hoppscotch version recently and the issue still exists there. Our theory is that there is a commit to the repository that has not been used in the public version of Hoppscotch yet.

Where should we go from here to resolve this issue?

**Comment by SanskritiHarmukh** [comment]
> @SanskritiHarmukh My team has hit a wall with this issue. We’ve successfully set up our build environments, however we haven’t been able to recreate the issue locally. We have been able to recreate it using the current version of Hoppscotch at https://hoppscotch.io/ though.
> 
> We’ve seen that there was an update to the Hoppscotch version recently and the issue still exists there. Our theory is that there is a commit to the repository that has not been used in the public version of Hoppscotch yet.
> 
> Where should we go from here to resolve this issue?

Will get back to you on this.

**Comment by SanskritiHarmukh** [comment]
Could you try setting up the Hoppscotch Agent as an interceptor and see if you can reproduce the issue on your end?

**Comment by SanskritiHarmukh** [comment]
Hi there,
I interacted with the team and we were able to figure out a few things. The issue seems to be related to the combination of the web app and the Agent. Specifically, when both are used together, the response is not being deflated properly. However, if you use just the web app, the response deflates automatically because the browser (via Axios) handles it. Similarly, if you're using the desktop app, the native interceptors are not deflating the response, which is where the issue also occurs.

The `hoppscotch-relay` library, which manages networking for both the desktop app and the Agent, is central to understanding this issue. Exploring how it interacts with the Agent could help address the problem.

If you're comfortable with Rust, there could be a potential solution involving libcurl to handle the deflation for us. If you’re interested in exploring this, you can dive into the rust-curl implementation and see how it can work with the current setup.

If Rust isn't your area, there's still an open issue for CodeDay that hasn't been addressed yet: [Hoppscotch Extension Issue #303](https://github.com/hoppscotch/hoppscotch-extension/issues/303). You can either look into that or explore other potential open issues. Let me know if you'd like to take one up.

**Comment by hissssst** [comment]
I experience this issue on hoppscotch electron desktop app

**Comment by xTudoS** [comment]
in my tests, hoppscotch don't accept-encoding headers when sending 

desktop app Version 25.1.1-0 (20250207.112242)
<img width="1574" alt="Image" src="https://github.com/user-attachments/assets/22ab14d7-65f9-4601-98d3-63255e6b51ea" />

https://hoppscotch.io
<img width="1533" alt="Image" src="https://github.com/user-attachments/assets/06b749ed-1e45-485c-ade1-d195a433fbc1" />


branch main localhost:

<img width="1786" alt="Image" src="https://github.com/user-attachments/assets/52628349-47ef-4684-a80b-d0e4fb95c851" />

**Comment by xTudoS** [comment]
Looks like to be something to do with the proxy, at least in the web version

https://github.com/user-attachments/assets/fd71cd55-a188-4f64-b0e1-e9797f9f7cf1

**Comment by xTudoS** [comment]
> Hi there, I interacted with the team and we were able to figure out a few things. The issue seems to be related to the combination of the web app and the Agent. Specifically, when both are used together, the response is not being deflated properly. However, if you use just the web app, the response deflates automatically because the browser (via Axios) handles it. Similarly, if you're using the desktop app, the native interceptors are not deflating the response, which is where the issue also occurs.
> 
> The `hoppscotch-relay` library, which manages networking for both the desktop app and the Agent, is central to understanding this issue. Exploring how it interacts with the Agent could help address the problem.
> 
> If you're comfortable with Rust, there could be a potential solution involving libcurl to handle the deflation for us. If you’re interested in exploring this, you can dive into the rust-curl implementation and see how it can work with the current setup.
> 
> If Rust isn't your area, there's still an open issue for CodeDay that hasn't been addressed yet: [Hoppscotch Extension Issue #303](https://github.com/hoppscotch/hoppscotch-extension/issues/303). You can either look into that or explore other potential open issues. Let me know if you'd like to take one up.

I find out where

adding curl_handle.accept_encoding("gzip");  to run_request_task solved

I will open a PR for it

**Comment by liyasthomas** [comment]
Thanks for your patience and valuable feedback. We’ve fixed this bug in the latest release.
[Download the latest builds](https://hoppscotch.com/download) and provide your valuable feedback.

Closing this ticket as this bug is now fixed in the recent version. Please feel free to reach out if you have any other concerns.
